R/pkg/R/sparkR.R

```diff
@@ -362,6 +362,10 @@ sparkR.session <- function(
   enableHiveSupport = TRUE,
   ...) {
 
+  if (length(sparkConfig[["spark.sql.warehouse.dir"]]) == 0) {
+    sparkConfig[["spark.sql.warehouse.dir"]] <- tempdir()
```

Comment:
As I mentioned in the JIRA, I think this change is not enough. If the user has a hive-site.xml and the warehouse dir set inside that, this change will override that [1]. We might need to change the Scala code and/or add a new option to specify that a new default for SparkR.

[1] https://github.com/apache/spark/blob/cf33a86285629abe72c1acf235b8bfa6057220a8/sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala#L49